sycl/include/CL/sycl/intel/sub_group_host.hpp

```diff
@@ -59,127 +59,126 @@ struct sub_group {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
-  template <typename T> T shuffle_up(T x, uint32_t delta) const {
+  template <typename T> T shuffle_up(T, uint32_t) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
 
-  template <typename T> T shuffle_xor(T x, id<1> value) const {
+  template <typename T> T shuffle_xor(T, id<1>) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
 
   /* --- two - input shuffles --- */
   /* indices in [0 , 2* sub - group size ) */
-  template <typename T> T shuffle(T x, T y, id<1> local_id) const {
+  template <typename T> T shuffle(T, T, id<1>) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
   template <typename T>
-  T shuffle_down(T current, T next, uint32_t delta) const {
+  T shuffle_down(T, T, uint32_t) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
   template <typename T>
-  T shuffle_up(T previous, T current, uint32_t delta) const {
+  T shuffle_up(T, T, uint32_t) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
 
   /* --- sub - group load / stores --- */
   /* these can map to SIMD or block read / write hardware where available */
   template <typename T, access::address_space Space>
-  T load(const multi_ptr<T, Space> src) const {
+  T load(const multi_ptr<T, Space>) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
 
   template <int N, typename T, access::address_space Space>
-  vec<T, N> load(const multi_ptr<T, Space> src) const {
+  vec<T, N> load(const multi_ptr<T, Space>) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
 
   template <typename T, access::address_space Space>
-  void store(multi_ptr<T, Space> dst, const T &x) const {
+  void store(multi_ptr<T, Space>, const T &) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
 
   template <int N, typename T, access::address_space Space>
-  void store(multi_ptr<T, Space> dst, const vec<T, N> &x) const {
+  void store(multi_ptr<T, Space>, const vec<T, N> &) const {
     throw runtime_error("Subgroups are not supported on host device. ",
                         PI_INVALID_DEVICE);
   }
 
   /* --- synchronization functions --- */
-  void barrier(access::fence_space accessSpace =
-                   access::fence_space::global_and_local) const {
+  void barrier(access::fence_space ) const {
```

Comment:
Should this one be `__UNUSED_ON_HOST(accessSpace)`?

A user should be free to call `barrier()` because of the default argument, but I think your change results in different definitions of this function for host and device.

Reply:
No. This is "host-only" defintions.
Device definition is located here: https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/intel/sub_group.hpp#L331

Reply:
But doesn't that mean kernels calling `barrier()` will fail to compile for the host device?  I thought we compiled all kernels for host and device.

Reply:
Sorry, you are right. I misunderstood the clarification from the first comment.